src/phoenix/trace/dsl/query.py

```diff
@@ -105,61 +87,155 @@ def from_dict(cls, obj: Mapping[str, Any]) -> "Projection":
 
 
 @dataclass(frozen=True)
-class Explosion(Projection):
+class _HasTmpSuffix(_Base):
+    _tmp_suffix: str = field(init=False, repr=False)
+    """Ideally every column name should get a temporary random suffix that will
+    be removed at the end. This is necessary during query construction because
+    sqlalchemy is not always foolproof, so we should actively avoid name
+    collisions, which is increasingly likely as queries get more complex. The
+    suffix is randomized per instance.
+    """
+
+    def __post_init__(self) -> None:
+        super().__post_init__()
+        object.__setattr__(self, "_tmp_suffix", f"{randint(0, 10**6):06d}")
+
+    def _remove_tmp_suffix(self, name: str) -> str:
+        if name.endswith(self._tmp_suffix):
+            return name[: -len(self._tmp_suffix)]
+        return name
+
+    def _add_tmp_suffix(self, name: str) -> str:
+        if name.endswith(self._tmp_suffix):
+            return name
+        return name + self._tmp_suffix
+
+
+@dataclass(frozen=True)
+class Explosion(_HasTmpSuffix, Projection):
     kwargs: Mapping[str, str] = field(default_factory=lambda: MappingProxyType({}))
```

Comment:
What do `kwargs` represent here in terms of the query?

Reply:
In `.explode("retrieval.documents", content="document.content", score="document.score")` the `kwargs` are `content=` and `score=`.

Reply:
it's the `kwargs` in `.explode(*args, **kwargs)`